psydac/core/bsplines.py

```diff
@@ -575,7 +578,7 @@ def elements_spans( knots, degree ):
 
     ie = 0
     for ik in range( degree, nk-degree ):
-        if knots[ik] != knots[ik+1]:
+        if knots[ik+1]-knots[ik]>=1e-15:
```

Comment:
See my previous comment to the `breakpoints` function: don't hide tolerances inside this function

Reply:
we had a conversation about this, we can either fix the tolerance to a number or add a kesword argument, that the user can change